src/displayapp/screens/StopWatch.cpp

```diff
@@ -123,97 +143,116 @@ void StopWatch::SetInterfaceStopped() {
-  SetInterfaceRunning();
-  startTime = xTaskGetTickCount();
-  currentState = States::Running;
-  wakeLock.Lock();
+void StopWatch::RenderTime() {
+  TimeSeparated elapsedTime = ConvertTicksToTimeSegments(stopWatchController.GetElapsedTime());
+  renderedSeconds = elapsedTime.epochSecs;
+  if (renderedSeconds.IsUpdated()) {
+    SetHoursVisible(elapsedTime.hours != 0);
+    if (!hoursVisible) {
+      lv_label_set_text_fmt(time, "%02d:%02d", elapsedTime.mins, elapsedTime.secs);
+    } else {
+      lv_label_set_text_fmt(time, "%02d:%02d:%02d", elapsedTime.hours, elapsedTime.mins, elapsedTime.secs);
+    }
+  }
+  lv_label_set_text_fmt(msecTime, "%02d", elapsedTime.hundredths);
 }
 
-void StopWatch::Pause() {
-  SetInterfacePaused();
-  startTime = 0;
-  // Store the current time elapsed in cache
-  oldTimeElapsed = laps[lapsDone];
-  blinkTime = xTaskGetTickCount() + blinkInterval;
-  currentState = States::Halted;
-  wakeLock.Release();
+void StopWatch::RenderPause() {
+  const TickType_t currentTime = xTaskGetTickCount();
+  if (currentTime > blinkTime) {
+    blinkTime = currentTime + blinkInterval;
+    if (lv_obj_get_state(time, LV_LABEL_PART_MAIN) == LV_STATE_DEFAULT) {
+      lv_obj_set_state(time, LV_STATE_DISABLED);
+      lv_obj_set_state(msecTime, LV_STATE_DISABLED);
+    } else {
+      lv_obj_set_state(time, LV_STATE_DEFAULT);
+      lv_obj_set_state(msecTime, LV_STATE_DEFAULT);
+    }
+  }
 }
 
-void StopWatch::Refresh() {
-  if (currentState == States::Running) {
-    laps[lapsDone] = oldTimeElapsed + xTaskGetTickCount() - startTime;
+void StopWatch::RenderLaps() {
+  lv_label_set_text(lapText, "");
+  for (int i = displayedLaps - 1; i >= 0; i--) {
+    std::optional<LapInfo> lap = stopWatchController.GetLapFromHistory(i);
 
-    TimeSeparated_t currentTimeSeparated = convertTicksToTimeSegments(laps[lapsDone]);
-    if (currentTimeSeparated.hours == 0) {
-      lv_label_set_text_fmt(time, "%02d:%02d", currentTimeSeparated.mins, currentTimeSeparated.secs);
-    } else {
-      lv_label_set_text_fmt(time, "%02d:%02d:%02d", currentTimeSeparated.hours, currentTimeSeparated.mins, currentTimeSeparated.secs);
-      if (!isHoursLabelUpdated) {
-        lv_obj_set_style_local_text_font(time, LV_LABEL_PART_MAIN, LV_STATE_DEFAULT, &jetbrains_mono_42);
-        lv_obj_realign(time);
-        isHoursLabelUpdated = true;
-      }
-    }
-    lv_label_set_text_fmt(msecTime, "%02d", currentTimeSeparated.hundredths);
-  } else if (currentState == States::Halted) {
-    const TickType_t currentTime = xTaskGetTickCount();
-    if (currentTime > blinkTime) {
-      blinkTime = currentTime + blinkInterval;
-      if (lv_obj_get_state(time, LV_LABEL_PART_MAIN) == LV_STATE_DEFAULT) {
-        lv_obj_set_state(time, LV_STATE_DISABLED);
-        lv_obj_set_state(msecTime, LV_STATE_DISABLED);
+    if (lap) {
+      TimeSeparated laptime = ConvertTicksToTimeSegments(lap->timeSinceStart);
+      char buffer[19];
+      if (laptime.hours == 0) {
```

Comment:
I think hours should be shown if the total runtime is >= 1 hour, rather than lap time >=1 hour. Having a leading zero in the hours column is nicely unambiguous. What do you think?

Reply:
Another style thought: since the time digits above having leading zeroes for minutes, I kinda feel like the lap numbers should too? But again it's a taste thing, I'll leave it up to you I'm happy either way

Reply:
Clicking around Wikipedia suggests lap times are generally written without leading zeroes (see [Nürburgring lap times](https://en.wikipedia.org/wiki/List_of_N%C3%BCrburgring_Nordschleife_lap_times#Lap_times) or [400-m sprint all-time bests](https://en.wikipedia.org/wiki/400_metres#All-time_top_25), for example). StopWatch was inconsistent about it, displaying minutes even for sub-minute laps but hiding hours until needed. I changed rendering to adhere to the convention and it now feels clean and tidy to me.